src/main/java/com/chemaxon/designhub/plugin/exampleimpl/PluginLogic.java

```diff
@@ -1,41 +1,38 @@
 package com.chemaxon.designhub.plugin.exampleimpl;
 
-import chemaxon.calculations.ElementalAnalyser;
-import chemaxon.formats.MolImporter;
-import chemaxon.struc.Molecule;
 import com.chemaxon.designhub.plugin.interfaces.RealtimePluginInterface;
 import com.chemaxon.designhub.plugin.interfaces.ResultSet;
 import com.chemaxon.designhub.plugin.settings.types.BooleanPluginSetting;
 import com.chemaxon.designhub.plugin.settings.types.NumberPluginSetting;
 import com.chemaxon.designhub.plugin.settings.types.StringPluginSetting;
+
+import chemaxon.calculations.ElementalAnalyser;
+import chemaxon.formats.MolImporter;
+import chemaxon.struc.Molecule;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
-import java.io.ByteArrayInputStream;
 import java.util.HashMap;
+import java.util.Map;
 
 @Component
 public class PluginLogic implements RealtimePluginInterface<PluginSettings> {
 
-    final PluginSettings myDefaultSettings = new PluginSettings(
+    private static final Logger logger = LoggerFactory.getLogger(PluginLogic.class);
+
+    private final PluginSettings myDefaultSettings = new PluginSettings(
             new BooleanPluginSetting("Enable report data", true),
             new StringPluginSetting("Put some nice string here", ""),
             new NumberPluginSetting("Custom atom count", 4));
 
-    ElementalAnalyser elementalAnalyser = new ElementalAnalyser();
-
-    final Logger logger = LoggerFactory.getLogger(PluginLogic.class);
-
-    public ResultSet getResultSet(String structure, String pinnedStructure, PluginSettings settings, Object context) {
+    public ResultSet getResultSet(String structure, String pinnedStructure, PluginSettings settings, Map<String, Object> context) {
         logger.debug("We just received some data!");
         try {
-            MolImporter mi = new MolImporter(new ByteArrayInputStream(structure.getBytes()));
-            Molecule mol = null;
-            mol = mi.read();
-
+            Molecule mol = MolImporter.importMol(structure);
+            ElementalAnalyser elementalAnalyser = new ElementalAnalyser();
```

Comment:
It needs to be instantiated for each request for thread-safety.
It might happen the instantiation is costly (this one is not), in that case we need to consider using resource pool.